### PR TITLE
Add CI and Fix Windows Build Errors

### PR DESCRIPTION
**Pull request by matthew-lidar**
## Summary of Changes

Add CI testing for each platform
Fix Windows include file and linker errors encountered
Fix a few tests on Windows

## Validation

Ran the added CI and local builds.

### CODE REVIEW

File: ouster_pcap/CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ target_include_directories(ouster_pcap PUBLIC
 CodeCoverageFunctionality(ouster_pcap)
 
 if(WIN32)
-  target_compile_options(ouster_pcap PRIVATE /wd4200)
+#  target_compile_options(ouster_pcap PRIVATE /wd4200)
```
Comment (matthew-lidar) [inline ouster_pcap/CMakeLists.txt:17]: Not sure what to do about this one. Am I good to remove it? It's breaking builds on mingw which work otherwise.

**Review by Samahu — APPROVED** [review]
looks good, i'd wait for the full test results before approving.

File: ouster_client/src/lidar_scan.cpp [hunk 2]
```diff
@@ -18,6 +18,10 @@
 #include "ouster/strings.h"
 #include "ouster/types.h"
 
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
```
[the remark this answers is not in the document]
Comment (matthew-lidar) [reply ouster_client/src/lidar_scan.cpp:24]: I tried that and it somehow didnt work

**Comment by Samahu** [comment]
Can you integrate this PR into yours? https://github.com/ouster-lidar/ouster_example/pull/598

**Review by Samahu — APPROVED** [review]

File: ouster_client/src/lidar_scan.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Samahu) [reply ouster_client/src/lidar_scan.cpp:24]: with the `_USE_MATH_DEFINES`?

File: ouster_pcap/CMakeLists.txt (hunk 1 above)
Comment (matthew-lidar) [reply ouster_pcap/CMakeLists.txt:17]: This warning isn't really removable: https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-levels-2-and-4-c4200?view=msvc-170
But we could silence in the code with a pragma instead if it still exists.

Comment (matthew-lidar) [reply ouster_pcap/CMakeLists.txt:17]: For now I'll just remove the warning suppression

File: ouster_client/src/lidar_scan.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (matthew-lidar) [reply ouster_client/src/lidar_scan.cpp:24]: Yeah, weirdly. I even tried swapping math headers with it and no luck.

**Review by akatumalla — APPROVED** [review]

**Review by akatumalla-ouster — APPROVED** [review]

**Pull request merged**